### DISCUSSION TITLE
Cicularbuffer in function

### DISCUSSION THREAD

**Discussion by bergeronnext** [rlogiacco/CircularBuffer#42] [Q&A] [answered]
Hi i can use the name buffer in fonction parameter 

like. Int  MyFonction (???? Buffer){

For(i=0,i<buffer.size(),i++){

Int avt +=Buffer[i] 

}
Return avt
}
My code is only a exemple
I dont know how to set the type of buffer

**Comment by rlogiacco** [comment, marked as answer]
First of all, your question is not clear at all, so my answer is my best guess of what you might have been asking.

My guess then is you want to pass an instance of CircularBuffer to a function, which you can obviously do:

```
long sum(CircularBuffer<int, 100> &cb) {
  long result = 0;
  for (int i = 0; i < cb.size(); i++)
    result += cb[i];
  return result;
}
```

You can call such function as `sum(buffer)` where `buffer` must be an instance of `CircularBuffer<int, 100>`.

You can define a _templated function_ if you put the following in a header file though:
```
template<size_t S>
long sum(CircularBuffer<int, S> &cb) {
  long result = 0;
  for (int i = 0; i < cb.size(); i++)
    result += cb[i];
  return result;
}
```

As you can see the size is now _variable_, so the `sum` function can work on any buffer of any size, given they are all storing `int` values.

Anyways, this is plain C++ programming, so please refer to a good C++ manual or tutorial covering templates.

**Reply by bergeronnext** [reply]
Thanks for your quick reply, that’s exactly what y wanted. Sorry for my  question which was imprecise.
